hail/src/main/scala/is/hail/services/package.scala

```diff
@@ -0,0 +1,53 @@
+package is.hail
+
+import is.hail.services.batch_client.ClientResponseException
+import org.apache.log4j.{LogManager, Logger}
+
+import scala.util.Random
+
+package object services {
+  lazy val log: Logger = LogManager.getLogger("is.hail.services")
+
+  val RETRYABLE_HTTP_STATUS_CODES: Set[Int] = {
+    val s = Set(408, 500, 502, 503, 504)
+    if (System.getenv("HAIL_DONT_RETRY_500") == "1")
+      s - 500
+    else
+      s
+  }
+
+  def sleepAndBackoff(delay: Double): Double = {
+    val t = delay * Random.nextDouble()
+    Thread.sleep((t * 1000).toInt)  // in ms
+    math.min(delay * 2, 60.0)
+  }
+
+  def isTransientError(e: Exception): Boolean = {
+    e match {
+      case e: ClientResponseException =>
+        RETRYABLE_HTTP_STATUS_CODES.contains(e.status)
+      case _ =>
+        false
+    }
+  }
+
+  def retryTransientErrors[T](f: => T): T = {
+    var delay = 0.1
+    var errors = 0
+    while (true) {
+      try {
+        return f
+      } catch {
+        case e: Exception =>
+          if (!isTransientError(e))
+            throw e
+          errors += 1
+          if (errors % 10 == 0)
+            log.warn(s"encountered $errors transient errors, most recent one was $e")
```

Comment:
subtle difference here, the python will log then tenth error even if it is not transient. I prefer the behavior here, can we change python to reflect this? I also prefer `if not is_transient_error(e)`, as done here.